spec.bs

```diff
@@ -1951,28 +1981,80 @@ Several APIs specific to fenced frames are defined on the {{Fence}} interface.
 
      1. Let |fencedFrameNonce| be |instance|'s [=fenced frame config instance/partition nonce=].
 
-     1. Let |credentiallessNonce| be
-
-     Issue: the page credentialless nonce
-     (<a href="https://github.com/WICG/fenced-frame/issues/191">WICG/fenced-frame#191</a>)
+     1. Let |credentiallessNonce| be |global|'s [=page credentialless nonce=].
 
      1. Invoke [=revoke network for a partition nonce=] on |fencedFrameNonce|.
 
-     1. Invoke [=revoke network for a partition nonce=] on |credentiallessNonce|.
-
-     1. Set |instance|'s [=fenced frame config instance/has disabled untrusted network=] to true.
+     1. If |credentiallessNonce| is non-null, invoke [=revoke network for a partition nonce=] on
+        |credentiallessNonce|.
 
-     1. Wait on all nested fenced frames to disable network too.
+     1. Set |instance|'s [=fenced frame config instance/untrusted network status=] to [=untrusted
+        network status/disabled for this tree=].
 
-     Issue: Spec this waiting more formally.
-     (<a href="https://github.com/WICG/fenced-frame/issues/151">WICG/fenced-frame#151</a>)
+     1. [=list/Append=] |p| to |instance|'s [=fenced frame config instance/on network disabled
+        promises=].
 
-     1. [=Queue a global task=] on the [=DOM manipulation task source=] given |global|, to
-        [=resolve=] |p| with {{undefined}}.
+     1. [=Recalculate the untrusted network status of all fenced frame descendants=] given
+        |global|'s [=Window/browsing context=]'s [=browsing context/top-level traversable=].
 
   1. Return |p|.
 </div>
 
+<div algorithm>
+  To <dfn>Recalculate the untrusted network status of all fenced frame descendants</dfn> given a
+  [=top-level traversable=] |topLevelTraversable|, run these steps:
+
+  1. [=Assert=]: this is running [=in parallel=].
+
+  1. Let |navigables| be |topLevelTraversable|'s [=navigable/active document=]'s
+     [=Document/inclusive descendant navigables=] with [=inclusive-dn-unfenced|unfenced=] set to
+     true.
+  
+  1. Let |navigablesWithNetworkChildren| be an empty [=set=].
```

Comment:
Yes, this is correct. The very last step in the algorithm (step 4.7.2) ensures that `|navigablesWithNetworkChildren|` will always contain fenced navigables. I wasn't sure if calling out that the set is expected to only contain certain items was a valid thing to do when declaring a variable in an algorithm directly, but since it sounds like it is, I can make that change.